### PR TITLE
206 replace safe sdk

### LINKED ISSUE #206: Replace safe.js logic with safe-core-sdk
Replace logic in safe.js using safe-core-sdk library and the new payment API with Gelato.

### PR DESCRIPTION
**Pull request by juanenrisley**
Closes #206

- [X] Replace functionality in safe.js to use `safe-core-sdk` and `circles-payment-api`
- [X] Create tests
- [X] Document all methods

### CODE REVIEW

**Comment by juanenrisley** [comment]
Tests are failing because they run against `circles-docker` and there is no `circles-payment-api` yet. If you run them in local, they pass.

File: src/safe.js [hunk 1]
```diff
@@ -127,448 +27,533 @@ export default function createSafeModule(
...
+        () => _getVersion(safeAddress),
+        (version) => version === SAFE_LAST_VERSION,
+        { label: 'Waiting for CRC Safe to upgrade version' },
+      );
+    }
+
+    return safeVersion;
+  };
+
+  // TODO: this method is missing to be implemented because it will be moved and replaced into organization.js.
```
Comment (JacqueGM) [inline src/safe.js:514]: should we remove it regardeless?

File: test/safe.test.js [hunk 2]
```diff
@@ -1,377 +1,190 @@
...
-
-    it('should recover the same Safe address when doing it again', async () => {
-      const safeAddressAgain = await core.safe.prepareDeploy(accounts[0], {
-        nonce,
-      });
-
-      expect(safeAddressAgain).toBe(safeAddress);
-    });
-
-    it('should be able to manually fund it for deployment', async () => {
```
Comment (JacqueGM) [inline test/safe.test.js:68]: this is also missing right - we said that we will enable this as it is now.

File: test/safe.test.js [hunk 3]
```diff
@@ -1,377 +1,190 @@
...
-      );
+    it('should get the safe address of the owner', () =>
+      core.safe
+        .getAddresses(accounts[0], {
+          ownerAddress: accounts[0].address,
+        })
+        .then((safeAddresses) => expect(safeAddresses).toContain(safeAddress)));
 
-      // Deploy Token as well to pay our fees later
-      await core.token.deploy(accounts[0], {
```
Comment (JacqueGM) [inline test/safe.test.js:97]: where are we doing the token deployment?

File: src/safe.js (hunk 1 above)
Comment (juanenrisley) [reply src/safe.js:514]: It is still there to remind that change in the futuro and do not forget.

File: test/safe.test.js (hunk 2 above)
Comment (juanenrisley) [reply test/safe.test.js:68]: It is still there to remind that change in the futuro and do not forget.

File: test/safe.test.js (hunk 3 above)
Comment (juanenrisley) [reply test/safe.test.js:97]: Nowhere, it was something not needed for these tests.

File: test/safe.test.js [hunk 4]
```diff
@@ -1,377 +1,190 @@
...
-      expect(status.isCreated).toBe(true);
-      expect(status.isDeployed).toBe(true);
-    });
+    it('should throw error when trying to deploy twice with same nonce', () =>
+      expect(() =>
+        core.safe.deploySafe(accounts[0], { nonce }),
+      ).rejects.toThrow(SafeDeployedError));
   });
 
-  describe('when a new Safe gets created through collecting trust connections', () => {
```
[the remark this answers is not in the document]
Comment (juanenrisley) [reply test/safe.test.js:113]: Yes, this is something to finish in the near future when trust functionality is available.

**Pull request merged**